.github/workflows/shake.yml

```
@@ -0,0 +1,61 @@
+# https://chat.openai.com/share/26102e95-ac9c-4d03-a000-73e4f6cee8cd
+name: Daily Shake Workflow
+
+on:
+  schedule:
+    - cron: '0 3 * * *'
+
+jobs:
+  shake-and-fix:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Run lake exe cache get
+        run: lake exe cache get
+
+      - name: Run lake exe shake --fix
+        run: env LEAN_ABORT_ON_PANIC=1 lake exe shake --fix
+
+      - name: Run lake build
+        run: lake build
+
+      - name: Check for local changes
+        id: git-check
+        run: |
+          if [[ $(git status --porcelain) ]]; then
+            echo "changes=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "changes=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Configure git user
+        if: ${{ steps.git-check.outputs.changes == 'true' }}
+        run: |
+          git config user.name "github-actions"
+          git config user.email "github-actions@github.com"
+
+      - name: Commit local changes and create PR
+        if: ${{ steps.git-check.outputs.changes == 'true' }}
+        run: |
+          DATE=$(date +%Y-%m-%d)
+          BRANCH="shake-$DATE"
+          git checkout -b "$BRANCH"
```

```suggestion
          git checkout -b "$BRANCH"  || exit 1
```
I do not know if this is needed, but I wonder what happens if the branch creation goes wrong -- probably not much anyway.